### PR TITLE
Ah gawb 3216: add logging for unique users who created a notebook

### PR DESCRIPTION
**Pull request by ahaessly**
add a log message with similar format to apache log messages for requests.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by ahaessly** [comment]
i'm hoping that since this is just logging, i can merge this to dev even with a dev freeze. more work will need to be done in logstash to identify this message

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -263,6 +263,50 @@ paths:
           description: Internal Error
           schema:
             $ref: '#/definitions/ErrorReport'
+  '/notebooks/{googleProject}/{clusterName}/api/contents':
```
Comment (rtitle) [inline src/main/resources/swagger/api-docs.yaml]: So this endpoint comes from Jupyter. Previously we've only been adding Swagger definitions for our Jupyter _extension_ APIs, not Jupyter APIs themselves. Jupyter has a lot more APIs which we don't document: https://github.com/jupyter/notebook/blob/master/notebook/services/api/api.yaml

My intuition is that we shouldn't be duplicating Jupyter's swagger inside Leo's swagger. I wonder if there's a way to "include" a swagger file from a different swagger file?

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala [hunk 2]
```diff
@@ -119,6 +119,9 @@ class ProxyService(proxyConfig: ProxyConfig,
           case Some(upgrade) => handleWebSocketRequest(targetHost, request, upgrade)
           case None => handleHttpRequest(targetHost, request)
         }
+        responseFuture map {response =>
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala]: I don't think the futures are hooked up correctly here. Should be something like:
```
        responseFuture map { response =>
          logRequestForMetrics(userInfo, request, response)
          response
        } recover { case e =>
          logger.error("Error occurred in Jupyter proxy", e)
          throw ProxyException(googleProject, clusterName)
        }
```

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala [hunk 3]
```diff
@@ -130,6 +133,20 @@ class ProxyService(proxyConfig: ProxyConfig,
     }
   }
 
+  private def logRequestForMetrics(userInfo: UserInfo, request: HttpRequest, response: HttpResponse) = {
+    val headers = request.headers
+    val headerMap: Map[String, String] = headers.map{ header =>
+      (header.name(),header.value())
+    }.toMap
+
+    logger.info(s"${headerMap.getOrElse("X-Forwarded-For", "0.0.0.0")} ${userInfo.userEmail} " +
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala]: I am wondering if the written will be in the right format. Our logback logger formats logs like so: https://github.com/DataBiosphere/leonardo/blob/develop/src/main/resources/logback.xml#L23

I wonder if we should have a separate `Logger` instance for access logs vs application logs.

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala [hunk 4]
```diff
@@ -130,6 +133,20 @@ class ProxyService(proxyConfig: ProxyConfig,
     }
   }
 
+  private def logRequestForMetrics(userInfo: UserInfo, request: HttpRequest, response: HttpResponse) = {
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala]: I wonder if this should be a directive in the route layer instead. It's kind of similar to `logRequestResult` with our own formatting applied: https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/debugging-directives/logRequestResult.html

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala (hunk 2 above)
Comment (ahaessly) [reply src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala]: right, thanks

**Pull request closed without merging**